sqs-lambda-ddb-cdk-ts/README.md

```diff
@@ -0,0 +1,129 @@
+# Offloading DynamoDB Writes with SQS, Lambda, and CDK
+
+The purpose of this pattern is to address write throttling issues of Provisioned Capacity DynamoDB Tables by offloading the requests to SQS to handle writes asynchronously. This pattern is deployed through an AWS Cloud Development Kit (CDK) app which provisions DynamoDB tables, SQS queues and DLQs, and Lambda functions. Test scripts have been provided for testing.
+
+Learn more about this pattern at Serverless Land Patterns: [INSERT SERVERLESS LAND LINK HERE]
+
+## Architecture Diagram
+
+![architecture_diagram](./images/architecture_diagram.png)
+
+## How it works
+
+A new Amazon SQS queue will be created and configured to invoke an AWS Lambda function that will write the received messages into an Amazon DynamoDB table. The SQS queue redrive policy will be configured to send undeliverable messages to an Amazon SQS Dead-Letter Queue. Undeliverable messages are the ones that couldn't be written to the DynamoDB table with a maximum of `maxReceiveCount` number of attempts.
+
+Lambda function will stop writing messages into the DynamoDB table as soon as it catches an exception and will report the remaining messages to Amazon SQS service as failed.
+
+Amazon DynamoDB table will have a minimal `ProvisionedThroughput` configuration for demo purposes, so that the users can observe message write retries when testing this stack. Real production throughput must be configured in accordance with the actual business requirements and message sizes.
```

Comment:
```suggestion
The Amazon DynamoDB table will have a minimal `ProvisionedThroughput` configuration for demo purposes, so that the users can observe retries on message writes when testing this stack. Real production throughput must be configured in accordance with the actual business requirements and message sizes.
```